internal/charmstore/store.go

```diff
@@ -711,35 +706,68 @@ func (s *Store) FindBestEntity(url *charm.URL, fields map[string]int) (*mongodoc
 		fields = nfields
 	}
 
+	errNotFound := errgo.WithCausef(nil, params.ErrNotFound, "entity not found")
+	findEntity := func(u *charm.URL, development bool) (*mongodoc.Entity, error) {
+		entity, err := s.FindEntity(&router.ResolvedURL{
+			URL:                 *u,
+			PromulgatedRevision: -1,
+			Development:         development,
+		}, fields)
+		if err != nil {
+			// A not found error is not contemplated here, as it would mean we
+			// have an internal inconsistency in which a base entity field
+			// refers to a non existing entity.
+			return nil, errgo.Notef(err, "cannot find charm or bundle for an internal error")
+		}
+		return entity, nil
+	}
+	if url.Revision == -1 {
+		// Find the current stable or development release.
+		baseEntity, err := s.FindBaseEntity(url, FieldSelector("developmentseries", "stableseries"))
```

Comment:
Getting the base entity here means we bypass the entitycache mechanism, which means the base entity won't be available in the cache and will need to be fetched again to check ACLs.